src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1481,30 +1481,20 @@ void uct_dc_mlx5_ep_pending_purge(uct_ep_h tl_ep,
 {
     uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_dc_mlx5_iface_t);
     uct_dc_mlx5_ep_t *ep       = ucs_derived_of(tl_ep, uct_dc_mlx5_ep_t);
+    uint8_t dci                = ep->dci;
     void *priv_args[2]         = {ep, arg};
     uct_purge_cb_args_t args   = {cb, priv_args};
     ucs_arbiter_t *waitq;
     ucs_arbiter_group_t *group;
     uint8_t pool_index;
-    uint8_t dci;
-
-    if (uct_dc_mlx5_iface_is_dci_rand(iface)) {
-        ucs_arbiter_group_purge(uct_dc_mlx5_iface_tx_waitq(iface),
-                                uct_dc_mlx5_ep_rand_arb_group(iface, ep),
-                                uct_dc_mlx5_ep_arbiter_purge_cb, &args);
-        return;
-    }
```

Comment:
in this case we may deschedule group for rand scheme on line 1511, which is not allowed

Reply:
we have a check for `uct_dc_mlx5_iface_is_dci_rand` below to return from the function